util/src/test/scala/org/ensime/util/DebouncerSpec.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.util
+
+import java.time.{ Clock, Instant, ZoneId }
+import org.scalatest._
+import scala.concurrent.duration._
+
+class DebouncerSpec extends FlatSpec with Matchers {
+  trait Fixture {
```

Comment:
the typical style has been to use the loan pattern to pass only the things that you want to make visible in the test, and taking config parameters... this exposes rather a lot of fields. Is that intentional?